crates/matrix-sdk/src/client/mod.rs

```diff
@@ -2371,4 +2384,21 @@ pub(crate) mod tests {
         let msc4028_enabled = client.can_homeserver_push_encrypted_event_to_device().await.unwrap();
         assert!(msc4028_enabled);
     }
+
+    #[async_test]
+    async fn test_delete_pusher() {
+        let server = MockServer::start().await;
+        let client = logged_in_client(Some(server.uri())).await;
+
+        Mock::given(method("POST"))
+            .and(path("_matrix/client/r0/pushers/set"))
+            .respond_with(ResponseTemplate::new(200).set_body_json(json!({})))
+            .mount(&server)
+            .await;
+
+        let pusher_ids = PusherIds::new("pushKey".to_owned(), "app_id".to_owned());
+        let response = client.delete_pusher(pusher_ids).await;
```

Comment:
We don't test that the `DELETE` has worked well, do we?

Reply:
no, not really, since it's just mocking the `POST` endpoint and asserting it's what being triggered on `delete_pusher`. so testing the unit of code in delete_pusher. I didn't find a manager that keeps track of pushers to check against for example. But please do suggest a different test case if you have it in mind :)

Reply:
Well, please, mock the `DELETE` verb on the same endpoint, and be sure it doesn't return a 404 for example.

Reply:
@Hywan I'm not sure if I got you correct, afaik there isn't a DELETE for this endpoint. The delete action is done by the POST endpoint when the kind is NULL. [ref](https://spec.matrix.org/v1.9/client-server-api/#post_matrixclientv3pushersset)

Reply:
Oh right! Sorry, I got confused!